src/DeviceInterface/ExtensionInterface.js

```diff
@@ -27,31 +27,56 @@ class ExtensionInterface extends InterfacePrototype {
             testMode: this.isTestMode()
         }
         const tooltipKinds = {
-            [POPUP_TYPES.EmailProtection]: 'legacy',
-            [POPUP_TYPES.EmailSignup]: 'emailsignup'
+            [TOOLTIP_TYPES.EmailProtection]: 'legacy',
+            [TOOLTIP_TYPES.EmailSignup]: 'emailsignup'
         }
-        const tooltipKind = tooltipKinds[this.getShowingTooltip()] || tooltipKinds[POPUP_TYPES.EmailProtection]
+        const tooltipKind = tooltipKinds[this.getShowingTooltip()] || tooltipKinds[TOOLTIP_TYPES.EmailProtection]
 
         return new HTMLTooltipUIController({ tooltipKind, device: this }, htmlTooltipOptions)
     }
 
-    get hasDismissedEmailSignup () {
-        // TODO -- implement peristed dismissed timestamp
-        return false
-    }
-
     getShowingTooltip () {
         if (this.hasLocalAddresses) {
-            return POPUP_TYPES.EmailProtection
+            return TOOLTIP_TYPES.EmailProtection
         }
 
-        if (this.settings.featureToggles.emailProtection_incontext_signup && !this.hasDismissedEmailSignup) {
-            return POPUP_TYPES.EmailSignup
+        if (this.settings.featureToggles.emailProtection_incontext_signup && this.settings.incontextSignupDismissed === false) {
+            return TOOLTIP_TYPES.EmailSignup
         }
 
         return null
     }
 
+    onIncontextSignupDismissed () {
+        this.settings.setIncontextSignupDismissed(true)
+        chrome.runtime.sendMessage({
+            messageType: 'setIncontextSignupDismissedAt',
+            options: {
+                value: new Date().getTime()
+            }
+        })
+        this.removeAutofillUIFromPage()
+    }
+
+    async resetAutofillUI () {
+        this.removeAutofillUIFromPage()
+
+        // Start the setup process again
+        await this.refreshSettings()
+        await this.setupAutofill()
+
+        // TODO: Do we need this to be called?
+        // await this.setupSettingsPage({shouldLog: true})
```

Comment:
I need to do a little more testing on this reset function -- it's working, but I'm seeing a validation error flash up on the OTP form just before the page refreshes. I'll need to follow the flow of what's happening in case we're doing a double submission or something -- and this settings page function may need called (though it was causing issues before).

Reply:
yeah we lack tests in general for the extension-based signin/signout etc.